modules/persistent-storage-csi-manila-dynamic-provisioning.adoc

```diff
@@ -0,0 +1,88 @@
+// Module included in the following assemblies:
+//
+// * storage/container_storage_interface/persistent-storage-csi-manila.adoc
+
+[id="persistent-storage-csi-manila-dynamic-provisioning_{context}"]
+= Dynamically provisioning CSI Manila volumes
+
+{product-title} installs a default StorageClass.
+
+As an application developer, you can dynamically provision RWX storage and deploy Pods with applications that safely consume the storage using YAML manifests. These YAML files are completely decoupled from Manila and from its CSI plug-in.
+
+You can use the same Pod and persistent volume claim (PVC) definitions on-premise that you use with {product-title} on AWS, GCP, Azure, and other platforms, with the exception of the storage class reference in the PVC definition.
+
+.Prerequisites
+
+* Storage must exist in the underlying infrastructure before it can be mounted as a volume in {product-title}.
+
+.UI procedure
+
+To dynamically create a CSI Manila volume using the web console:
+
+. In the {product-title} console, click *Storage* → *Persistent Volume Claims*.
+
+. In the persistent volume claims overview, click *Create Persistent Volume Claim*.
+
+. Define the required options on the resulting page.
+
+.. Select the `csi-manila-ceph` StorageClass.
+
+.. Enter a unique name for the storage claim.
+
+.. Select the access mode to specify read and write access for the PVC you are creating.
++
+[IMPORTANT]
+====
+* Use RWX if you want the persistent volume (PV) that fulfills this PVC to be mounted to multiple Pods on multiple nodes in the cluster.
+
+* Use RWO mode if you want to prevent additional Pods from being dynamically provisioned.
+====
+
+. Define the size of the storage claim.
+
+. Click *Create* to create the PersistentVolumeClaim and generate a PersistentVolume.
+
+.CLI procedure
+
+To dynamically create a CSI Manila volume using the command-line interface:
+
+. Create and save a file with the PersistentVolumeClaim object described by the following YAML:
+
+
++
+.pvc-manila.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: myclaim
+spec:
+  accessModes: <1>
+    - ReadWriteMany
+  resources:
+    requests:
+  	storage: 10Gi
+  storageClassName: csi-manila-nfs <2>
+----
++
+<1> Use RWX if you want the persistent volume (PV) that fulfills this PVC to be mounted to multiple Pods on multiple nodes in the cluster. Use RWO mode to prevent additional Pods from being dynamically provisioned.
+<2> The name of the storage class that provisions the storage back end. This was created by the cluster administrator.
++
+. Create the object you saved in the previous step by running the following command:
++
+----
+$ oc create -f pvc-manila.yaml
+----
++
+A new PVC is created.
+
+. Verify that the volume was created and is ready by running the following command:
```

Comment:
```suggestion
. To verify that the volume was created and is ready, run the following command:
```

Reply:
I agree with your previous comment that there's not much difference here. I like your construction better here ("run the following command" sounds better than "by running"), so I'll update this line.